src/hardware/serialport/serialdummy.cpp

```diff
@@ -78,25 +87,32 @@ void CSerialDummy::transmitByte(Bit8u val, bool first) {
 /*****************************************************************************/
 
 void CSerialDummy::setBreak(bool value) {
-	//LOG_MSG("UART 0x%x: Break toggeled: %d", base, value);
+	(void)value; // unused
+	// LOG_MSG("SERIAL: Port %u at UART 0x%x break "
+	//         "toggeled: %d.", GetPortNumber(), base, value);
 }
 
 /*****************************************************************************/
 /* setRTSDTR sets the modem control lines                                   **/
 /*****************************************************************************/
-void CSerialDummy::setRTSDTR(bool rts, bool dtr) {
-	setRTS(rts);
-	setDTR(dtr);
+void CSerialDummy::setRTSDTR(bool rts_state, bool dtr_state)
+{
+	setRTS(rts_state);
+	setDTR(dtr_state);
 }
 void CSerialDummy::setRTS(bool val) {
 #ifdef CHECKIT_TESTPLUG
 	setCTS(val);
+#else
+	(void)val; // unused
```

Comment:
I don't think we need this :)

GCC warns about unused function parameters only with -Wextra (-Wunused-parameter to be exact); the more important -Wunused-variable is included in -Wall already.

When C++17 will happen we could use standard `[[maybe_unused]]` attribute. 

Perhaps we could introduce MAYBE_UNUSED macro to wrap gcc's and clang's [`unused`](https://gcc.gnu.org/onlinedocs/gcc-3.2/gcc/Function-Attributes.html) attribute. Theoretically we could use `[[gnu:unused]]` right now, but I think it won't compile in MSVC (not sure though). Overall I'm not sure if old `unused` attribute works for parameters, but C++17 `maybe_unused` definitely does.

Using attributes is better way to deal with these warnings (if you really, really need to; personally I don't think we need, because they are in -Wextra).

edit
Just tested and hey - in GCC attributes seem to *not* be tied to the standard used for compilation (!). So perhaps we can start using it right away; I'll investigate further.

Reply:
Thanks for the detailed explanation and your thoughts.  Always learning something! Will hold off, and then apply your suggestion to the handful of unused variables we've got in this PR!

Reply:
Let's leave this be for future PR, focusing only on introducing "unused".